src/tests/test_dynamic_api/test_views_wfs.py

```diff
@@ -180,3 +124,72 @@ def test_wfs_non_default_dataset_not_exposed(
         )
         response = api_client.get(wfs_url)
         assert response.status_code == 404
+
+
+@pytest.mark.django_db
+def test_wfs_model_auth(api_client, geometry_authdataset_thing, fetch_auth_token, filled_router):
+    url = (
+        "/v1/wfs/geometry_authdataset/"
+        "?SERVICE=WFS&VERSION=2.0.0&REQUEST=GetFeature&TYPENAMES=things"
+        "&OUTPUTFORMAT=application/gml+xml"
+    )
+    # We need both scopes TEST/TOP[12].
+    token = fetch_auth_token(["TEST/TOP1"])
+
+    response = api_client.get(url, HTTP_AUTHORIZATION=f"Bearer {token}")
+    assert response.status_code == 403
+
+    token = fetch_auth_token(["TEST/TOP1", "TEST/TOP2"])
+
+    response = api_client.get(url, HTTP_AUTHORIZATION=f"Bearer {token}")
+    assert response.status_code == 200
+
+    # We should get a full result, regardless of "auth" on properties,
+    # if we have access to the dataset.
+    xml_root = read_response_xml(response)
+    data = xml_element_to_dict(xml_root[0][0])
+    assert data == {
+        "boundedBy": None,
+        "geometry": None,
+        "id": "1",
+        "metadata": "secret",
+    }
+
+
+@pytest.mark.django_db
+def test_wfs_field_auth(api_client, geometry_auth_thing, fetch_auth_token, filled_router):
+    def get_response(*scopes):
```

Comment:
It is good practice to avoid any complexity in tests, and better to split test into 3 different tests instead of reusing `get_response` function inside.

Splitting test into 3 different functions would make tests more readable and also will help isolating issues in code a bit easier. 
Single  use case tests in combination clear names for tests save ton of time when debugging broken code.

Reply:
Maybe split up the different testcases using `pytest.parametrize`.

Reply:
This is indeed a good case for `pytest.mark.parametrize`.

Or
* move the `get_response()` outside the function
* have 3 separate test calls. 
* all grouped in a single class